source/filter.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const map = {
+    '&': '&amp;',
+    '<': '&lt;',
+    '>': '&gt;',
+    '"': '&quot;',
+    "'": '&#39;'
+  };
+
+const filter = (text, tags = []) => {
+    if(!(typeof text === "string")){
+        throw new Error("expected String");
+    }
+    if (!Array.isArray(tags)){
+        throw new Error("expected Array");
+    }
+    let resArr = text.split(/([&"<>'])/g);
+    for(let i = 0; i < resArr.length; i++){
+        if(map[resArr[i]] !== undefined){
+            if( resArr[i] == "<" && resArr[i+2] == ">" &&
+                tags.includes(resArr[i+1].startsWith("/") ? resArr[i+1].slice(1) : resArr[i+1])
```

Comment:
Хочу извращений в угоду меньшего объема кода =)
```suggestion
                tags.includes(resArr[i+1].slice(+resArr[i+1].startsWith("/")))
```